packages/fuels-signers/src/provider.rs

```diff
@@ -60,6 +63,92 @@ impl From<TimeParameters> for FuelTimeParameters {
     }
 }
 
+pub(crate) struct ResourceQueries {
+    utxos: Vec<String>,
+    messages: Vec<String>,
+    asset_id: String,
+    amount: u64,
+}
+
+impl ResourceQueries {
+    pub fn new(
+        utxo_ids: Vec<UtxoId>,
+        message_ids: Vec<MessageId>,
+        asset_id: AssetId,
+        amount: u64,
+    ) -> Self {
+        let utxos = utxo_ids
+            .iter()
+            .map(|utxo_id| format!("{utxo_id:#x}"))
+            .collect::<Vec<_>>();
+
+        let messages = message_ids
+            .iter()
+            .map(|msg_id| format!("{msg_id:#x}"))
+            .collect::<Vec<_>>();
+
+        Self {
+            utxos,
+            messages,
+            asset_id: format!("{asset_id:#x}"),
+            amount,
+        }
+    }
+
+    pub fn exclusion_query(&self) -> Option<(Vec<&str>, Vec<&str>)> {
+        if self.utxos.is_empty() && self.messages.is_empty() {
+            return None;
+        }
+
+        let utxos_as_str = self.utxos.iter().map(AsRef::as_ref).collect::<Vec<_>>();
+
+        let msg_ids_as_str = self.messages.iter().map(AsRef::as_ref).collect::<Vec<_>>();
+
+        Some((utxos_as_str, msg_ids_as_str))
+    }
+
+    pub fn spend_query(&self) -> Vec<(&str, u64, Option<u64>)> {
+        vec![(self.asset_id.as_str(), self.amount, None)]
+    }
+}
+
+// ANCHOR: resource_filter
+pub struct ResourceFilter {
+    pub from: Bech32Address,
+    pub asset_id: AssetId,
+    pub amount: u64,
+    pub excluded_utxos: Vec<UtxoId>,
+    pub excluded_message_ids: Vec<MessageId>,
+}
+// ANCHOR_END: resource_filter
+
+impl ResourceFilter {
+    pub fn owner(&self) -> String {
+        self.from.hash().to_string()
+    }
+
+    pub(crate) fn resource_queries(&self) -> ResourceQueries {
```

Comment:
Having the methods on `ResourceFilter ` would require us to keep the fields of `ResourceQueries` inside it. 
This downgrades the devx since a user can't initialise a filter conveniently like this:

```rust
let filter = ResourceFilter {
   amount,
   asset_id,
   ..Default::default()
}
```